cpp/src/common/cumlHandle_impl.hpp

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2019-2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <mutex>
+#include <unordered_map>
+#include <utility>
+#include <vector>
+
+#include <cublas_v2.h>
+#include <cusolverDn.h>
+#include <cusolverSp.h>
+#include <cusparse.h>
+
+#include <common/cuml_comms_int.hpp>
+
+#include <cuml/cuml.hpp>
+
+#include <cuml/common/cuml_allocator.hpp>
+
+#include "handle_impl.hpp"
+
+namespace ML {
+
+using MLCommon::deviceAllocator;
+using MLCommon::hostAllocator;
+
+/**
+ * @todo: Add doxygen documentation
```

Comment:
Looks like this is even from the existing `cumlHandle.hpp`. Can you create a Github issue for this? If we're going to keep this abstraction around for some time, it's useful to have it documented.